sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -95,7 +103,16 @@ def Execute(client, global_endpoint_manager, function, *args, **kwargs):
             client.last_response_headers[
                 HttpHeaders.ThrottleRetryWaitTimeInMs
             ] = resourceThrottle_retry_policy.cumulative_wait_time_in_milliseconds
-
+            # If container does not have throughput, results will return empty list, but to align with other SDKs
```

Comment:
Aligning with other SDKs is not a priority for the Python SDK - idiomatic Python is the priority.
Is this code resulting in a change in customer facing behaviour?
I see we've moved some of these artificial errors out of `container.py` - does that mean they were refactored here as part of the new retry implementation and customer behaviour remains unchanged?

If not, can you point me to a sample/test/code snippet that demonstrates the current GA behaviour and what you're proposing to change it to?

Reply:
This still returns a 404 with the same message as before, the only difference is the addition of a python sdk specific sub status code to help with container recreate retry and that it errors out here so we can retry it in the case of container recreate.

Reply:
IMO I'd have such an exception generation piece of logic to where the `Offer` resource is queried - `_retry_utility.py` should simply decide based on the exception what the appropriate retry policy should be.

Reply:
The original code had this in `get_throughput` in `_container.py` which actually makes more sense than leaking the abstraction into `_retry_utility.py`:

```python
        if len(throughput_properties) == 0:
            raise CosmosResourceNotFoundError(
                status_code=StatusCodes.NOT_FOUND,
                message="Could not find Offer for container " + self.container_link)
```

Reply:
We artificially/manually raise the exception, but doing so in the container class would make it so we can't use retry logic as that is done right before we return the empty query result in get_throughput.

Reply:
I kind agree with Abhijeet, get_throughput might be the better place to raise the specific error codes.  Is it possible to change to let retry_utility wrap around get_throughput (or maybe adding an internal method for this purpose)?